src/wannierise.F90

```diff
@@ -1878,14 +1885,19 @@ subroutine wann_domega(csheet,sheet,rave,cdodq)
                 cdodq_loc(m,n,nkp_loc) = cdodq_loc(m,n,nkp_loc) &
                      + wb(nn) * 0.5_dp &
                      *( cr(m,n) - conjg(cr(n,m)) )
-                ! -S[T^{k,b}]=-(T+Tdag)/2i ; T_mn = Rt_mn q_n
-                cdodq_loc(m,n,nkp_loc) = cdodq_loc(m,n,nkp_loc) -  &
+                ! +(lambdac-1)S[T^{k,b}]-lambdac S[Tc^{k,b}]
+                !S[T] = (T+Tdag)/2i ; T_mn = Rt_mn q_n; Tc_mn = Rt_mn qc_n
+                !q_n^{k,b} = Im ln M_nn^{k,b} + b.r_n; qc_n^{k,b} = Im ln M_nn^{k,b} + b.r0
+                cdodq_loc(m,n,nkp_loc) = cdodq_loc(m,n,nkp_loc) - & 
                       ( crt(m,n) * ln_tmp_loc(n,nn,nkp_loc)  &
                      + conjg( crt(n,m) * ln_tmp_loc(m,nn,nkp_loc) ) ) &
                      * cmplx(0.0_dp,-0.5_dp,kind=dp)
-                cdodq_loc(m,n,nkp_loc) = cdodq_loc(m,n,nkp_loc) - wb(nn) &
+                cdodq_loc(m,n,nkp_loc) = cdodq_loc(m,n,nkp_loc) + wb(nn) * (lambdac - 1.0_dp) &
                      * ( crt(m,n) * rnkb_loc(n,nn,nkp_loc) + conjg(crt(n,m) &
                      * rnkb_loc(m,nn,nkp_loc)) ) * cmplx(0.0_dp,-0.5_dp,kind=dp)
+                cdodq_loc(m,n,nkp_loc) = cdodq_loc(m,n,nkp_loc) - lambdac * wb(nn) &
+                     * ( crt(m,n) * r0kb(nn,nkp_loc) + conjg(crt(n,m) &
+                     * r0kb(nn,nkp_loc)) ) * cmplx(0.0_dp,-0.5_dp,kind=dp)
```

Comment:
I think there is something not right in calling r0kb with nkp_loc, either here - or earlier when the data is setup. It doesn't appear consistent in the two places.

Reply:
Hi, sorry about this. The commit for for the selectively localised WFs shouldn't be merged to the main repository yet. Not sure what is the best way to proceed now.